File: client/src/models/Classification.js
import { APIItem } from "./APIItem.js";

export class Classification extends APIItem {
  constructor(data) {
    super(data)
    this.title = data.title
    this.description = data.description
    this.cryptidImg = data.cryptidImg
    this.terrainImg = data.terrainImg
  }

  get terrainBackgroundImgUrl() {
    return `url(${this.terrainImg})`
  }
}

export class CryptidClassificationClassification extends Classification {
  constructor(data) {
    super(data)
    this.cryptidClassificationId = data.cryptidClassificationId
  }
}

File: server/Models/Classification.cs
using System.ComponentModel.DataAnnotations;

namespace cryptipedia.Models;

public class Classification : RepoItem<int>
{
  public string Title { get; set; }
  [MaxLength(5000)]
  public string Description { get; set; }
  [Url, MaxLength(3000)]
  public string CryptidImg { get; set; }
  [Url, MaxLength(3000)]
  public string TerrainImg { get; set; }
}